python/llm/dev/benchmark/harness/run_multi_llb.py

```diff
@@ -0,0 +1,163 @@
+            model=args.model,
+            model_args=model_args,
+            tasks=task_names,
+            num_fewshot=num_fewshot,
+            batch_size=args.batch_size,
+            max_batch_size=args.max_batch_size,
+            device=device,
+            no_cache=args.no_cache,
+            limit=args.limit,
+            description_dict=description_dict,
+            decontamination_ngrams_path=args.decontamination_ngrams_path,
+            check_integrity=args.check_integrity,
+            write_out=args.write_out,
+            output_base_path=log_dir
+        )
+        if len(results['results']) > 1:
+            average = {}
+            for _, subtask in results['results'].items():
+                for metric, value in subtask.items():
+                    average[metric] = average.get(metric, []) + [value]
+            for k, v in average.items():
+                average[k] = sum(v) / len(v) if not k.endswith("_stderr") else 0
+            results['results'][task] = average
+            results['versions'][task] = 1
+
+        dumped = json.dumps(results, indent=2)
+        print(dumped)
+
+        if args.output_path:
+            with open(f"{log_dir}/result.json", "w") as f:
+                f.write(dumped)
+        result_pool.put(results)
+        device_pool.put(device)
+    
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model", required=True)
+    parser.add_argument("--model_args", default="")
+    parser.add_argument("--pretrained", required=True, type=str)
+    parser.add_argument("--tasks", required=True, nargs='+', type=str)
+    parser.add_argument("--precision", required=True, nargs='+', type=str)
+    parser.add_argument("--provide_description", action="store_true")
+    parser.add_argument("--num_fewshot", type=int, default=0)
+    parser.add_argument("--batch_size", type=str, default=None)
+    parser.add_argument(
+        "--max_batch_size",
+        type=int,
+        default=None,
+        help="Maximal batch size to try with --batch_size auto",
+    )
+    parser.add_argument("--device", type=str, default=None)
+    parser.add_argument("--output_path", default=None)
+    parser.add_argument(
+        "--limit",
+        type=float,
+        default=None,
+        help="Limit the number of examples per task. "
+        "If <1, limit is a percentage of the total number of examples.",
+    )
+    parser.add_argument("--data_sampling", type=float, default=None)
+    parser.add_argument("--no_cache", action="store_true")
+    parser.add_argument("--decontamination_ngrams_path", default=None)
+    parser.add_argument("--description_dict_path", default=None)
+    parser.add_argument("--check_integrity", action="store_true")
+    parser.add_argument("--write_out", action="store_true", default=False)
+    parser.add_argument("--output_base_path", type=str, default=None)
+
+    return parser.parse_args()
+
+
+def main():
+    mp.set_start_method('spawn')
```

Comment:
This is required. Otherwise processes will be hanged or the memory won't be released.